### ISSUE TITLE
AuthorizationPolicy update didn't push to sidecar when there is a global Sidecar CRD

### ISSUE THREAD

**Issue by zhaoli2333** [istio/istio#32420]
**Bug description**
I create a global Sidecar like:
```
apiVersion: networking.istio.io/v1alpha3
kind: Sidecar
metadata:
  name: default
  namespace: istio-system
spec:
  egress:
    - hosts:
        - "./*"
        - "istio-system/*"
```

then i create an AuthorizationPolicy like this:
```
apiVersion: security.istio.io/v1beta1
kind: AuthorizationPolicy
metadata:
  namespace: istio-demo
spec:
  action: ALLOW
  rules:
  - to:
    - operation:
        paths:
        - /*
    when:
    - key: request.auth.claims[resource_access][istio-demo][roles]
      values:
      - xxx_user
  selector:
    matchLabels:
      app: istio-demo-app
```

This  AuthorizationPolicy is in the same namespace with my deployment, but when i send requests to the application , the policy did not work, unless I restart the pod or delete the global Sidecar config.

And I checked the config_dump file of the sidecar  and tried to find something like "envoy.filters.http.rbac", but it wasn't changed when I updated the AuthorizationPolicy.

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**

**Steps to reproduce the bug**

**Version** 
1.8.4

**How was Istio installed?**
Istio Operator

**Environment where the bug was observed (cloud vendor, OS, etc)**

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

**Comment by howardjohn** [comment]
I could not reproduce on `master`.

**Comment by zhaoli2333** [comment]
> I could not reproduce on `master`.

will it be fixed in 1.8.x？
